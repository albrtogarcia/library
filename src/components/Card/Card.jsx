import PropTipes from 'prop-types'
import { Icon } from '../Icon/Icon.jsx'
import './Card.scss'

export const Card = ({ title, icon, children }) => (
  <div className="card">
    <div className="card-header">
      <Icon name={icon} />
      <h3 className="card-title">{title}</h3>
    </div>
    <div className="card-body">{children}</div>
  </div>
)

Card.propTypes = {
  title: PropTipes.string.isRequired,
  icon: PropTipes.string.isRequired,
  children: PropTipes.node.isRequired
}
