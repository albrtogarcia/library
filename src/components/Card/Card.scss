.card {
  display: flex;
  flex-direction: column;
  border-radius: var(--legacyDisplay, 1rem);
  border: 1px solid var(--border, #E2E8F0);
  background: var(--white, #FFF);

  &-header {
    display: flex;
    padding: 1rem 1.5rem;
    align-items: center;
    gap: 0.5rem;
    align-self: stretch;
  }
}