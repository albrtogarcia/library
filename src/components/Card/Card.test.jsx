import { describe, expect, it } from 'vitest'
import { Card } from './Card'

describe('Card', () => {
  const props = {
    title: 'Test',
    children: 'Test',
    icon: 'Test'
  }

  it('matches snapshot', () => {
    const element = Card(props)
    expect(element).toMatchSnapshot()
  })
})
