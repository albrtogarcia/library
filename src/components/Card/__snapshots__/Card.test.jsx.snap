// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > matches snapshot 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    <Icon
      name="Test"
    />
    <h3
      className="card-title"
    >
      Test
    </h3>
  </div>
  <div
    className="card-body"
  >
    Test
  </div>
</div>
`;
